device/common/src/get_prefix_sum.cpp

```diff
@@ -19,15 +19,14 @@ prefix_sum_t get_prefix_sum(
     vecmem::memory_resource& mr) {
 
     // Create the result object.
-    prefix_sum_t result(&mr);
     const std::size_t nelements = static_cast<std::size_t>(
         std::accumulate(sizes.begin(), sizes.end(), 0));
-    result.reserve(nelements);
+    prefix_sum_t result(nelements, &mr);
 
     // Fill the result object.
-    for (std::size_t i = 0; i < sizes.size(); ++i) {
-        for (std::size_t j = 0; j < sizes[i]; ++j) {
-            result.push_back({i, j});
+    for (std::size_t i = 0, k = 0; i < sizes.size(); ++i) {
+        for (std::size_t j = 0; j < sizes[i]; ++j, ++k) {
+            result[k] = {i, j};
```

Comment:
Let's do something fancy. Let's add:

```c++
assert(k < nelements);
```

before this line. That way assuring that debug builds would test the assumption that we're making in the code. (You will also need to include `<cassert>` for this.)

Reply:
You know what, let's make that instead:

```c++
assert(k < result.size());
```

Since the whole thing is not there in an optimised build, we should not even assume what the size of the `result` object is. :wink: